.claude/hooks/post-tool-call.py

```
@@ -0,0 +1,93 @@
+PORT_FILE_SUFFIX = "-provenance-port.txt"
+
+class ProvenanceHookError(RuntimeError):
+    pass
+
+def http_request(method, host, port, location, *, body: Optional[bytes] = None, headers={}, timeout=None) -> bytes:
+    with closing(HTTPConnection(host, port, timeout=timeout)) as connection:
+        connection.request(method, location, body=body, headers=headers)
+
+def get_server_port():
+    claude_root = os.getenv("CLAUDE_PROJECT_DIR")
+    path_hash = hashlib.md5(claude_root.encode('utf-8')).hexdigest()
```

Ruff (0.14.14)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)